src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,132 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.exception.LoadingError;
+import duke.exception.SavingError;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Class that handles laoding and saving tasks to a file.
+ */
+public class Storage {
+
+    private final String filepath;
+
+    /**
+     * Storage constructor.
+     *
+     * @param filepath The filepath of the task list from the project root
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Returns an ArrayList<Task> of loaded Tasks from the file.
+     *
+     * @return ArrayList<Task> of tasks
+     * @throws DukeException that indicates what the issue was in opening/parsing the file
+     */
+    public ArrayList<Task> load() throws DukeException {
+        String home = System.getProperty("user.dir");
+        Path path = Paths.get(home, this.filepath);
+        boolean directoryExists = Files.exists(path);
```

Comment:
Maybe can name as isDirectoryFound for it to look more like a boolean?

Reply:
> Overall, namings are easy to read and understand, good job!
> You may want to add more javadoc for some public methods for reader to understand them better.

Thank you for the useful input! I've made the necessary changes.